### PR TITLE
Show new error message when output schema is not maching input

### PR DESCRIPTION
**Pull request by bajram-adapt**
Jira: https://issues.cask.co/browse/PLUGIN-132?workflowName=jira&stepId=3
Original PR: https://github.com/data-integrations/dynamic-spark/pull/33/files

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -100,6 +100,11 @@
       <version>4.11</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>io.cdap.plugin</groupId>
+      <artifactId>hydrator-common</artifactId>
+      <version>2.4.0-SNAPSHOT</version>
```
Comment (CuriousVini) [inline pom.xml:106]: This should be provided scope

Comment (CuriousVini) [inline pom.xml:106]: Please add it as a property and not hardcode

File: pom.xml [hunk 2]
```diff
@@ -22,14 +22,14 @@
...
-  <version>2.2.2</version>
+  <version>2.3.0-SNAPSHOT</version>
 
   <properties>
     <!-- properties for script build step that creates the config files for the artifacts -->
     <data.pipeline.parent>system:cdap-data-pipeline[6.0.0-SNAPSHOT,7.0.0-SNAPSHOT)</data.pipeline.parent>
     <data.stream.parent>system:cdap-data-streams[6.0.0-SNAPSHOT,7.0.0-SNAPSHOT)</data.stream.parent>
 
-    <cdap.version>6.1.3-SNAPSHOT</cdap.version>
+    <cdap.version>6.2.0</cdap.version>
```
Comment (CuriousVini) [inline pom.xml]: why is this change needed?

File: pom.xml [hunk 3]
```diff
@@ -22,14 +22,14 @@
 
   <groupId>io.cdap.plugin</groupId>
   <artifactId>dynamic-spark</artifactId>
-  <version>2.2.2</version>
+  <version>2.3.0-SNAPSHOT</version>
```
Comment (CuriousVini) [inline pom.xml]: This should be `2.2.3`

**Review by CuriousVini — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (bajram-adapt) [reply pom.xml]: 6.2.0 was the version in the merged PR the cherrypick is from? Is this incorrect and needs to be changed

File: pom.xml (hunk 3 above)
Comment (CuriousVini) [reply pom.xml]: This still should be fixed

File: pom.xml (hunk 2 above)
Comment (CuriousVini) [reply pom.xml]: Also lets make this 6.1.1

File: pom.xml [hunk 4]
```diff
@@ -22,14 +22,14 @@
...
 
   <properties>
     <!-- properties for script build step that creates the config files for the artifacts -->
     <data.pipeline.parent>system:cdap-data-pipeline[6.0.0-SNAPSHOT,7.0.0-SNAPSHOT)</data.pipeline.parent>
```
Comment (CuriousVini) [inline pom.xml:29]: Change this to 6.1.1

Comment (bajram-adapt) [reply pom.xml:29]: change what to 6.1.1? the pipeline and stream . parents ?

Comment (bajram-adapt) [reply pom.xml:29]: should we go back and make the same changes in develop as well since this is a cherrypick?

**Review by CuriousVini — APPROVED** [review]

File: pom.xml (hunk 4 above)
Comment (CuriousVini) [reply pom.xml:29]: yea we can keep them same in develop as well as release. 
`[61.1,7.0.0-SNAPSHOT)`

File: pom.xml (hunk 1 above)
Comment (CuriousVini) [inline pom.xml]: Also lets not use snapshot version in release branch. Make it a property with `2.4.0` version

**Review by CuriousVini — APPROVED** [review]
lgtm
